wallet/src/repository/wallet/state.rs

```diff
@@ -103,17 +109,49 @@ impl State {
         self.transient_internal_addresses.clear();
         self.transient_external_addresses.clear();
     }
+}
 
-    /// Tell whether the wallet is undergoing a synchronization.
-    ///
-    /// Currently, this function derives the information from the presence or absence of transient
-    /// addresses, i.e. addresses that are temporarily generated during the synchronization process.
-    ///
-    /// TODO: refactor all synchronization-related fields into a `SynchronizationState` structure
-    ///  so that the number of fields in `State` does not keep growing, and take that as a chance
-    ///  to add a private `is_syncing` field for which this would act as a getter.
+/// The synchronization state and information for a wallet.
+#[derive(Debug)]
+pub struct SynchronizationStatus {
+    is_syncing: bool,
+    is_resyncing: bool,
+}
+
+impl SynchronizationStatus {
+    /// Tell whether a wallet is syncing.
+    #[allow(dead_code)]
```

Comment:
```suggestion
```

Reply:
Are you suggesting removing or commenting out the function?

Reply:
Removing, public functions cannot be dead code. If clippy complains when you remove it may be because it's not actually public, it may be inside a private module or something. In that case you can keep the allow(dead_code).

Reply:
I marked it as dead code only because I believe we may use it in the future, but just not yet. My take is that `impl` for data structures are let's say "library code", so, to the greatest extent possible, their API shouldn't be conditioned by usage.